pints/_nested/_nestedEllipsoid.py

```diff
@@ -0,0 +1,313 @@
+            if i < self._rejectionSamples:
+                mActive[aMinIndex,:] = rejectSamplePrior(aRunningLogLikelihood,self._log_likelihood,self._prior)
+            else:
+                mActive[aMinIndex,:] = rejectEllipsoidSample_faster(aRunningLogLikelihood,self._log_likelihood,mActive[:,:d],self._enlargementFactor,A,centroid)
+
+        vLogZ[self._iterations] = logsumexp(mActive[:,d])
+        w[self._iterations:] = float(X[self._iterations]) / float(self._activePoints)
+        mSamplesAll = np.vstack((mInactive,mActive))
+        logZ = logsumexp(vLogZ, b = w[0:(self._iterations+1)])
+        vLogP = mSamplesAll[:,d] - logZ + np.log(w)
+        vP = np.exp(mSamplesAll[:,d] - logZ) * w
+        mTheta = mSamplesAll[:,:-1]
+        vIndex = np.random.choice(range(0,(self._iterations + self._activePoints)), self._numPosteriorSamples, p=vP)
+        mPosteriorSamples = mTheta[vIndex,:]
+        return [mPosteriorSamples, logZ]
+
+    def set_activePoints_rate(self, activePoints):
+        """
+        Sets the number of active points for the next run.
+        """
+        activePoints = float(activePoints)
+        if activePoints <= 5:
+            raise ValueError('Number of active points must be greater than 5.')
+        self._activePoints = activePoints
+
+    def set_iterations(self, iterations):
+        """
+        Sets the total number of iterations to be performed in the next run
+        (including burn-in and non-adaptive iterations).
+        """
+        iterations = int(iterations)
+        if iterations < 0:
+            raise ValueError('Number of iterations cannot be negative.')
+        self._iterations = iterations
+
+    def set_numPosteriorSamples(self,numPosteriorSamples):
+        
+        if numPosteriorSamples > np.round(0.25 * (self._iterations + self._activePoints)):
+            raise ValueError('Number of posterior samples must be fewer than 25% the total number of preminary points')
+        self._numPosteriorSamples = numPosteriorSamples
+
+    def set_enlargementFactor(self,enlargementFactor):
+        if enlargementFactor < 1:
+            raise ValueError('Enlargement factor must exceed 1')
+        self._enlargementFactor = enlargementFactor
+
+    def set_rejectionSamples(self,rejectionSamples):
+        if rejectionSamples < 0:
+            raise ValueError('Must have non-negative rejection samples')
+        if rejectionSamples > self._iterations:
+            raise ValueError('Must have fewer rejection samples than total samples')
+        self._rejectionSamples = rejectionSamples
+    def set_ellipsoidUpdateGap(self,ellipsoidUpdateGap):
+        if ellipsoidUpdateGap < 1:
+            raise ValueError('Ellipsoid update gap must exceed 1')
+        if isinstance(ellipsoidUpdateGap,int) == False:
+            raise ValueError('Ellipsoid update gap must be an integer')
+        self._ellipsoidUpdateGap = ellipsoidUpdateGap
+
+
+def rejectDrawFromEllipsoid(A,centroid,aLogLikelihood,aThreshold):
```

Comment:
- Should these be methods rather than stand-alone functions? (just asking!)
- I take it these aren't called by users? If so, please use the convention `_hidden_stuff_starts_with_an_underscore` for naming them
- reject_draw_from_ellipsoid